pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotEvaluateLiteralRule.java

```diff
@@ -258,13 +275,18 @@ private static Object convertResultValue(@Nullable Object resultValue, RelDataTy
         return TimestampUtils.toMillisSinceEpoch(resultValue.toString());
       }
     }
-    // Return BigDecimal for numbers
-    if (resultValue instanceof Integer || resultValue instanceof Long) {
+    // Use BigDecimal for INTEGER / BIGINT / DECIMAL literals
```

Comment:
In SSE, we use int for INTEGER and long for BIGINT, should we keep it consistent?

Reply:
Hm, I followed this logic to keep it consistent with Calcite which uses `BigDecimal` for literals of these SQL types. I think it makes more sense to try to align with Calcite rather than SSE as long as we're not breaking anything here?

Reply:
In this particular case though, I don't think it should make much of a difference, so I'm happy to update this if you'd like.

Reply:
Either way is fine. We can discuss the long term plan to make SSE and MSE consistent separately

Reply:
I've updated this because I realized that even though Calcite uses `BigDecimal` for these literals, we do convert it into `int` / `long` during rel to plan conversion before sending it into our execution engine - https://github.com/apache/pinot/blob/1fdeadd6ef9ad3c6ca51f198ac0d4f922e92a0b7/pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpressionUtils.java#L237-L245